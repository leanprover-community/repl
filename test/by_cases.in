{"cmd": "theorem foo (x : Int) : x = x := by sorry"}

{"proofState" : 0, "tactic": "by_cases h : x < 0"}

{"proofState" : 1, "tactic": "case inl => rfl"}

{"proofState" : 1, "tactic": "all_goals sorry"}

