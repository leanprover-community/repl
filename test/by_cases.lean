theorem foo (x : Int) : x = x := by
  by_cases h : x < 0
  next => rfl
  next => rfl
