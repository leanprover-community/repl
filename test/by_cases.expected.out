{"sorries":
 [{"proofState": 0,
   "pos": {"line": 1, "column": 36},
   "goal": "x : Int\n⊢ x = x",
   "endPos": {"line": 1, "column": 41}}],
 "messages":
 [{"severity": "warning",
   "pos": {"line": 1, "column": 8},
   "endPos": {"line": 1, "column": 11},
   "data": "declaration uses 'sorry'"}],
 "env": 0}

{"proofState": 1,
 "goals":
 ["case inl\nx : Int\nh : x < 0\n⊢ x = x",
  "case inr\nx : Int\nh : ¬x < 0\n⊢ x = x"]}

{"proofState": 2, "goals": ["case inr\nx : Int\nh : ¬x < 0\n⊢ x = x"]}

{"sorries":
 [{"proofState": 3, "goal": "case inl\nx : Int\nh : x < 0\n⊢ x = x"},
  {"proofState": 4, "goal": "case inr\nx : Int\nh : ¬x < 0\n⊢ x = x"}],
 "proofState": 5,
 "goals": []}

